manager.py

```diff
@@ -28,8 +28,8 @@ def remove_worker(conn, host):
     worker_dict = ({'host': host, 'port': 5432})
 
     print("removing %s" % host, file=stderr)
-    cur.execute("""DELETE FROM pg_dist_shard_placement WHERE nodename=%(host)s AND
-                                                             nodeport=%(port)s;
+    cur.execute("""DELETE FROM pg_dist_placement WHERE groupid = (SELECT groupid FROM 
```

Comment:
Even though the host and port _should_ uniquely identify a node, it would probably be more "correct" if you used a `WHERE groupid IN (/* subquery */)` rather than saying `groupid =`, since there isn't an inherent guarantee that subquery will always return one row.

That all said, perhaps slapping on a `LIMIT 1` would accomplish the same. None of this is super-critical, it's just that it would be nicer to somehow make explicit the fact that the subquery could return more than one row at a future date.

Reply:
done